Sources/Leaf/LeafEncoder.swift

```diff
@@ -90,12 +86,8 @@ private final class SingleValueContainer: SingleValueEncodingContainer, _Contain
 
     /// See `SingleValueEncodingContainer`
     func encode<T>(_ value: T) throws where T: Encodable {
-        if let convertible = value as? LeafDataRepresentable {
-            if let converted = convertible.leafData {
-                self.data = .data(converted)
-            } else {
-                throw EncodingError.invalidValue(value, at: self.codingPath)
-            }
+        if let leafRepresentable = value as? LeafDataRepresentable {
+            self.data = .data(leafRepresentable.leafData)
```

Comment:
Remove conditional assignment from change to `LeafDataRepresentable.leafData: LeafData`